lib/models/os.ts

```diff
@@ -0,0 +1,674 @@
+	 * Unsupported (unpublished) version will result in rejection.
+	 * The version **must** be the exact version number.
+	 * To resolve the semver-compatible range use `balena.model.os.getMaxSatisfyingVersion`.
+	 * @fulfil {Date} - last modified date
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.os.getLastModified('raspberry-pi').then(function(date) {
+	 * 	console.log('The raspberry-pi image was last modified in ' + date);
+	 * });
+	 *
+	 * balena.models.os.getLastModified('raspberrypi3', '2.0.0').then(function(date) {
+	 * 	console.log('The raspberry-pi image was last modified in ' + date);
+	 * });
+	 *
+	 * balena.models.os.getLastModified('raspberry-pi', function(error, date) {
+	 * 	if (error) throw error;
+	 * 	console.log('The raspberry-pi image was last modified in ' + date);
+	 * });
+	 */
+	const getLastModified = function(
+		deviceType: string,
+		version: string = 'latest',
+		callback?: (error?: Error, result?: Date) => void,
+	): Promise<Date> {
+		callback = findCallback(arguments);
+
+		return getValidatedDeviceType(deviceType)
+			.then(() => normalizeVersion(version))
+			.then(ver =>
+				request.send({
+					method: 'HEAD',
+					url: deviceImageUrl(deviceType, ver),
+					baseUrl: apiUrl,
+				}),
+			)
+			.catch(notFoundResponse, () => {
+				throw new Error('No such version for the device type');
+			})
+			.then(response => new Date(response.headers.get('last-modified')!))
+			.asCallback(callback);
+	};
+
+	/**
+	 * @summary Download an OS image
+	 * @name download
+	 * @public
+	 * @function
+	 * @memberof balena.models.os
+	 *
+	 * @param {String} deviceType - device type slug
+	 * @param {String} [version] - semver-compatible version or 'latest', defaults to 'latest'
+	 * Unsupported (unpublished) version will result in rejection.
+	 * The version **must** be the exact version number.
+	 * To resolve the semver-compatible range use `balena.model.os.getMaxSatisfyingVersion`.
+	 * @fulfil {ReadableStream} - download stream
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.models.os.download('raspberry-pi').then(function(stream) {
+	 * 	stream.pipe(fs.createWriteStream('foo/bar/image.img'));
+	 * });
+	 *
+	 * balena.models.os.download('raspberry-pi', function(error, stream) {
+	 * 	if (error) throw error;
+	 * 	stream.pipe(fs.createWriteStream('foo/bar/image.img'));
+	 * });
+	 */
+	const download = onlyIf(!isBrowser)(function(
+		deviceType: string,
+		version: string = 'latest',
+		callback?: (error?: Error, result?: BalenaRequestStreamResult) => void,
+	): Promise<BalenaRequestStreamResult> {
+		callback = findCallback(arguments);
+
+		return getValidatedDeviceType(deviceType)
+			.then(() => {
+				if (version === 'latest') {
+					return _getOsVersions(deviceType).get('latest');
+				}
+
+				return normalizeVersion(version);
+			})
+			.then(ver =>
+				request.stream({
+					method: 'GET',
+					url: deviceImageUrl(deviceType, ver),
+					baseUrl: apiUrl,
```

Comment:
Let's add back the comment that explains why this is not omitting the token, so that we don't forget about it.
> optionally authenticated, so we send the token in all cases